engine/client/filesync.go

```diff
@@ -262,8 +262,10 @@ func (t FilesyncTarget) DiffCopy(stream filesync.FileSend_DiffCopyServer) (rerr
 		return fmt.Errorf("failed to create synctarget dest file %s: %w", finalDestPath, err)
 	}
 	defer destF.Close()
-	if err := destF.Chown(int(t.uid), int(t.gid)); err != nil {
-		return fmt.Errorf("failed to chown synctarget dest file %s: %w", finalDestPath, err)
+	if runtime.GOOS != "windows" {
```

Comment:
LGTM, I have a broader question on this (I'm still learning go idioms)

in `.NET world` you would have smth like abstract class and maybe `WindowsDriver` or `LinuxDriver` and then delegate to make sure all code in those implementation is unique to that OS but here we are having same execution regardless of OS with if branch.

In go it is possible still to do how I describe or is it more common in golang with this approach?

Reply:
We can use build constraint to implement chown in specific platforms. 

https://pkg.go.dev/cmd/go#hdr-Build_constraints